.github/workflows/miden.yml

```diff
@@ -0,0 +1,210 @@
+name: EC2 Miden Node Setup and Upgrade
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Deployment Environment'
+        required: true
+        default: 'testnet'
+
+      additional_aws_role:
+        description: 'ARN of the role to assume in the additional AWS account'
+        required: true
+
+  push:
+    branches:
+      - next
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  setup-ec2:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::211125515935:role/Miden-GithubActionsRole
+          aws-region: us-west-1
+
+    - name: Configure AWS credentials for additional account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::657556092833:role/Miden-GithubActionsRole
+          aws-region: us-west-1
```

Comment:
I suggest we use a single job, and use an env variable to distinguish between testnet and devnet. The steps can then use this variable to do the conditional work. What do you think about something like the following:

```suggestion
      - name: Select testnet
        if: github.event_name == 'workflow_dispatch'
        run: echo "NETWORK=testnet" >> $GITHUB_ENV
      
      - name: Select devnet
        if: github.event_name != 'workflow_dispatch'
        run: echo "NETWORK=devnet" >> $GITHUB_ENV
        
      - name: Configure testnet AWS credentials
        if: env.NETWORK == 'testnet'
        uses: aws-actions/configure-aws-credentials@v1
        with:
          role-to-assume: arn:aws:iam::211125515935:role/Miden-GithubActionsRole
          aws-region: us-west-1

      - name: Configure devnet AWS credentials
        if: env.NETWORK == 'devnet'
        uses: aws-actions/configure-aws-credentials@v1
        with:
          role-to-assume: arn:aws:iam::657556092833:role/Miden-GithubActionsRole
          aws-region: us-west-1
```
and then we can similarly also configure the ip to use:
```
      - name: Configure testnet IP
        if: env.NETWORK == 'testnet'
        run: echo "AWS_IP=54.154.193.162" >> $GITHUB_ENV

      - name: Configure devnet IP
        if: env.NETWORK == 'devnet'
        run: echo "AWS_IP=3.255.177.27" >> $GITHUB_ENV
```
so that we can combine things:
```
      - name: Install prerequisites via SSH
        run: |
          ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@{{ env.AWS_IP }} << 'EOF'
            sudo yum update -y
            sudo yum install -y git gcc openssl-devel bzip2-devel libffi-devel make
            sudo yum groupinstall -y "Development Tools"
            # Install Rust
            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
            source $HOME/.cargo/env
            rustup update
            # Install dependencies
            sudo yum install -y clang cmake
EOF
```
And also a separate step for installing the binary with `cargo install` for devnet and testnet.